pull.go

```diff
@@ -229,6 +240,11 @@ func (c *Client) fetch(ctx context.Context, rCtx *RemoteContext, ref string, lim
 }
 
 func (c *Client) createNewImage(ctx context.Context, img images.Image) (images.Image, error) {
+	span := trace.SpanFromContext(ctx)
```

Comment:
This will be the parent of the `Span` created in `Pull()` above, won't it?  Since the new context created alongside `pullSpan` is discarded the `fetchImageSpan` created here will be a sibling of `pullSpan` instead of a child, which is what I would expect the relationship to be.

Reply:
To get the correct `Span`, the `context.Context` returned by `Tracer.Start()` (in `Pull()`) would need to be passed here?

Reply:
Yes, `Tracer.Start()` puts the newly created `Span` in the `context.Context` it returns, so that new `Context` should be passed to any functions down the call stack.

Reply:
Let me check that part and experiment. Thanks for pointing out.